app/Controllers/extensionController.php

```diff
@@ -175,30 +187,36 @@ public function disableAction() {
 				Minz_Request::bad(_t('feedback.extensions.not_enabled', $ext_name), $url_redirect);
 			}
 
+			$type = $ext->getType();
+			if ($type !== 'user' && !FreshRSS_Auth::hasAccess('admin')) {
+				Minz_Request::bad(_t('feedback.extensions.no_access', $ext_name), $url_redirect);
+				return;
+			}
+
 			$conf = null;
-			if ($ext->getType() === 'system' && FreshRSS_Auth::hasAccess('admin')) {
+			if ($type === 'system') {
 				$conf = FreshRSS_Context::$system_conf;
-			} elseif ($ext->getType() === 'user') {
+			} elseif ($type === 'user') {
 				$conf = FreshRSS_Context::$user_conf;
-			} else {
-				Minz_Request::bad(_t('feedback.extensions.no_access', $ext_name), $url_redirect);
 			}
 
 			$res = $ext->uninstall();
 
 			if ($res === true) {
 				$ext_list = $conf->extensions_enabled;
-				$legacyKey = array_search($ext_name, $ext_list, true);
-				if ($legacyKey !== false) {	//Legacy format FreshRSS < 1.11.1
```

Comment:
Removed support for 5 year-old configuration syntax for FreshRSS < 1.11.1.
Automatically update by just saving the configuration